sql/src/main/java/io/crate/planner/operators/ExplainLogicalPlan.java

```diff
@@ -0,0 +1,199 @@
+        }
+
+        @Override
+        protected ImmutableMap.Builder<String, Object> visitPlan(LogicalPlan logicalPlan, Context context) {
+            return toMap(logicalPlan, subMap());
+        }
+
+        @Override
+        public ImmutableMap.Builder<String, Object> visitMultiPhase(MultiPhase logicalPlan, Context context) {
+            Map<String, Map<String, Object>> dependencies = new HashMap<>(logicalPlan.dependencies().size());
+            for (Map.Entry<LogicalPlan, SelectSymbol> entry : logicalPlan.dependencies().entrySet()) {
+                dependencies.put(entry.getValue().representation(), explainMap(entry.getKey(), context));
+            }
+            return toMap(logicalPlan, subMap()
+                .put("source", explainMap(logicalPlan.source, context))
+                .put("dependencies", dependencies));
+        }
+
+        @Override
+        public ImmutableMap.Builder<String, Object> visitFetchOrEval(FetchOrEval logicalPlan, Context context) {
+            return toMap(logicalPlan, subMap()
+                .put("fetchRefs", refs(logicalPlan.outputs()))
+                .put("source", explainMap(logicalPlan.source, context)));
+        }
+
+        @Override
+        public ImmutableMap.Builder<String, Object> visitCollect(Collect logicalPlan, Context context) {
+            return toMap(logicalPlan, subMap()
+                .put("toCollect", refs(logicalPlan.outputs()))
+                .put("where", logicalPlan.where.query().representation()));
+        }
+
+        @Override
+        public ImmutableMap.Builder<String, Object> visitLimit(Limit logicalPlan, Context context) {
+            return toMap(logicalPlan, subMap()
+                .put("limit", SymbolPrinter.INSTANCE.print(logicalPlan.limit, SymbolPrinter.Style.FULL_QUALIFIED))
+                .put("offset", SymbolPrinter.INSTANCE.print(logicalPlan.offset, SymbolPrinter.Style.FULL_QUALIFIED))
+                .put("source", explainMap(logicalPlan.source, context)));
+        }
+
+        @Override
+        public ImmutableMap.Builder<String, Object> visitCount(Count logicalPlan, Context context) {
+            return toMap(logicalPlan, subMap()
+                .put("where", logicalPlan.where.query().representation()));
+        }
+
+        @Override
+        public ImmutableMap.Builder<String, Object> visitGet(Get logicalPlan, Context context) {
+            return toMap(logicalPlan, subMap()
+                .put("toCollect", refs(logicalPlan.outputs()))
+                .put("docKeys", logicalPlan.docKeys.toString()));
+        }
+
+        @Override
+        public ImmutableMap.Builder<String, Object> visitFilter(Filter logicalPlan, Context context) {
+            return toMap(logicalPlan, subMap()
+                .put("filter", logicalPlan.queryClause.query().representation())
+                .put("source", explainMap(logicalPlan.source, context)));
```

Comment:
I wonder if we could improve the method naming a bit.  The difference between `toMap` and `explainMap` is a bit unclear